lib/denormalizers/upvote_given.rb

```diff
@@ -0,0 +1,24 @@
+module Denormalizers
+  class UpvoteGiven
+    class CannotPlusOneYourself < StandardError; end
+    class InvalidSlackToken < StandardError; end
+
+    def call(event)
+      return if Upvote.find_by(uid: event.data[:upvote_uid]).present?
+      ActiveRecord::Base.transaction do
+        team = Team.find(event.data[:team_id])
+        sender, recipient = prepare_transaction_actors(team).call(event.data[:params])
+        raise InvalidSlackToken if recipient.slack_user_name == 'u'
```

Comment:
Add this logic should be in aggregate class. No event should be published because this command could not be executed.